arrow/arrow.py

```diff
@@ -314,7 +318,14 @@ def strptime(cls, date_str, fmt, tzinfo=None):
     # factories: ranges and spans
 
     @classmethod
-    def range(cls, frame, start, end=None, tz=None, limit=None):
+    def range(
+        cls,
+        frame: str,
+        start: Union["Arrow", dt_datetime],
+        end: Union["Arrow", dt_datetime, None] = None,
+        tz: Union[tzfile, str, None] = None,
+        limit: Union[int, None] = None,
+    ) -> Union[Iterator[Union[Iterator, Iterator["Arrow"]]], "Arrow"]:
```

Comment:
This is an interesting one. Any explanation as to why this is so complicated? Doesn't range just return an `Iterator["Arrow"]`?